FWCore/Framework/interface/Principal.h

```diff
@@ -218,16 +217,9 @@ namespace edm {
     //called by adjustIndexesAfterProductRegistryAddition only if an index actually changed
     virtual void changedIndexes_() {}
 
-    void addScheduledProduct(std::shared_ptr<BranchDescription const> bd);
-    void addSourceProduct(std::shared_ptr<BranchDescription const> bd);
+    //called by adjustIndexesAfterProductRegistryAddition
     void addDelayedReaderInputProduct(std::shared_ptr<BranchDescription const> bd);
```

Comment:
Will work on a separate PR to remove the code that depends on these last two methods.